src/test/java/org/reaktivity/nukleus/tcp/internal/streams/ClientIT.java

```diff
@@ -237,4 +379,48 @@ public void shouldInitiateClientClose() throws Exception
             }
         }
     }
+
+    @Test
+    @Specification({
+        "${route}/output/new/controller",
+        "${streamsInvalid}/client.sent.data.close.data/client/source"
+    })
+    public void shouldResetIfDataReceivedAfterEndOfStream() throws Exception
+    {
+        try (ServerSocket server = new ServerSocket())
+        {
+            server.setReuseAddress(true);
+            server.bind(new InetSocketAddress("127.0.0.1", 0x1f90));
+            server.setSoTimeout((int) SECONDS.toMillis(5));
+
+            k3po.start();
+            k3po.awaitBarrier("ROUTED_OUTPUT");
+
+            try (Socket socket = server.accept())
+            {
+                k3po.notifyBarrier("ROUTED_INPUT");
+
+                final InputStream in = socket.getInputStream();
+
+                byte[] buf = new byte[256];
+                int len = in.read(buf);
+
+                assertEquals("client data", new String(buf, 0, len, UTF_8));
+
+                try
+                {
+                    len = in.read(buf);
+                }
+                catch (IOException ex)
+                {
+                    System.out.println(ex);
+                    len = -1;
```

Comment:
I am assuming it's a legitimate code path since the behavior when the other end resets the TCP connection is not very clear from the SocketChannel.read javadoc. I would imagine that it would be legitimate  for the implementation to throw an IOException in this case.